src/Orleans/Storage/SqlFormatProvider.cs

```diff
@@ -0,0 +1,83 @@
+﻿/*
+Project Orleans Cloud Service SDK ver. 1.0
+ 
+Copyright (c) Microsoft Corporation
+ 
+All rights reserved.
+ 
+MIT License
+
+Permission is hereby granted, free of charge, to any person obtaining a copy of this software and 
+associated documentation files (the ""Software""), to deal in the Software without restriction,
+including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense,
+and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so,
+subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS
+OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*/
+
+using System;
+using System.Globalization;
+
+
+namespace Orleans.Runtime.Storage.Relational
+{
+    /// <summary>
+    /// Formats .NET types appropriately for database consumption in non-parameterized queries.
+    /// </summary>
+    public class SqlFormatProvider: IFormatProvider
+    {
+        private readonly SqlFormatter formatter = new SqlFormatter();
+
+
+        public object GetFormat(Type formatType)
+        {
+            if(formatType == typeof(ICustomFormatter))
+            {
+                return formatter;
+            }
+
+            return null;
+        }
+
+
+        private class SqlFormatter: ICustomFormatter
```

Comment:
A formatter to format parameter to SQL queries in a well-defined matter in one place. This is needed especially in cases whe ``DbParameter`` objects are not used (Orleans uses currently parameters everywhere). One place to use these currently would be in tracing the queries.